gazebosim/gz-cmake
# Changes for OGRE-2.3

# 🎉 New feature

## Summary

CMake updates to make `gz-cmake` compatible with OGRE-2.3 debs built by OpenRobotics

## Test it


For linux (jammy and focal):
* Download the latest ogre-2.3 debs either from the prerelease mirror or by using the direct links from Jenkins: https://build.osrfoundation.org/job/ogre-2.3-debbuilder/27/
* Build gz-cmake

For homebrew:
* Use the new OGRE 2.3 bottle: https://github.com/osrf/homebrew-simulation/pull/1971

```
cd gz-cmake/examples/find_ogre2/ogre-2.3
mkdir build && cd build
cmake ..
```

The expected output with the debs installed

```
-- Looking for GzOGRE2 - found

-- OGRE2_FOUND: TRUE
-- OGRE2_LIBRARIES: /usr/lib/x86_64-linux-gnu/OGRE-2.3/libOgreNextMain.so
-- OGRE2_INCLUDE_DIRS: /usr/include/OGRE-2.3/usr/include/OGRE-2.3/RenderSystems/GL3Plus
-- OGRE2_VERSION: 2.3.1
-- OGRE2_VERSION_MAJOR: 2
-- OGRE2_VERSION_MINOR: 3
-- OGRE2_VERSION_PATCH: 1
-- OGRE2_RESOURCE_PATH: /usr/lib/x86_64-linux-gnu/OGRE-2.3/OGRE
-- Configuring done
-- Generating done
-- Build files have been written to: /home/mjcarroll/workspaces/gz_garden/src/gz-cmake/examples/find_ogre2/ogre-2.3/build
```

Similarly, the `ogre-2.2` demo should continue to find `2.2.5` on `jammy`.


## Checklist

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.
